### PR TITLE
Don’t crash app if no elastic search is available

### PR DESCRIPTION
**Pull request by tkasper**
The app should not crash if there is no connection to the elastic search

### CODE REVIEW

**Comment by apfelbox** [comment]
I don't think that this is the only place where this exception can be thrown? I thought that the issue also occurs when people are trying to search? That isn't this code path.

**Comment by apfelbox** [comment]
You need both, of course, as the exception can be thrown in multiple places.

**Comment by apfelbox** [comment]
Ah, I just saw that you added it in the client directly – which is the correct place. 👍 
However, you missed the `bulk` call further below.

**Comment by apfelbox** [comment]
Can you also apply the attached patch, please?

[0001-Cleanup-code.patch.zip](https://github.com/Becklyn/SearchBundle/files/2995942/0001-Cleanup-code.patch.zip)

File: Search/SearchIndexer.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 use Becklyn\SearchBundle\Metadata\Metadata;
 use Becklyn\SearchBundle\Metadata\MetadataFactory;
 use Doctrine\Common\Util\ClassUtils;
+use Elasticsearch\Common\Exceptions\NoNodesAvailableException;
```
Comment (apfelbox) [inline Search/SearchIndexer.php]: ⚰️ ?

File: Elasticsearch/ElasticsearchClient.php [hunk 2]
```diff
@@ -115,12 +120,13 @@ public function sendRequests (array $requests)
...
         $currentBulk = [];
         $maxIndex = count($requests) - 1;
 
-        for ($i = 0; $i <= $maxIndex; $i++)
+        foreach ($requests as $i => $request)
         {
-            $request = $requests[$i];
             $data = $request->getData();
 
             // add header
```
Comment (apfelbox) [inline Elasticsearch/ElasticsearchClient.php]: The `$this->client->bulk()` call in line 147 is still not protected from throwing exceptions.

**Review by apfelbox — CHANGES_REQUESTED** [review]

**Comment by tkasper** [comment]
Oh, sorry, didn't see the bulk one. Fixed it.

**Review by apfelbox — APPROVED** [review]

**Pull request merged**